core/modules/file/file.pages.inc

```diff
@@ -498,6 +498,13 @@ function file_manage_form($form, &$form_state, File $file) {
     );
   }
 
+  $form['name'] = array(
+    '#type' => 'item',
+    '#title' => t('Filename'),
+    '#markup' => $file->filename,
```

Comment:
Looks likely to possibly create an XSS issue here. Wrapping in a `check_plain()` wouldn't hurt.